packages/contracts/contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol

```diff
@@ -971,75 +973,97 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
     }
 
     /**
-     * Checks that a given batch context has a time context which is below a given que element
-     * @param _context The batch context to validate has values lower.
-     * @param _queueIndex Index of the queue element we are validating came later than the context.
+     * For a fair of
+     * @param _earlierContext The batch context to validate has values lower.
+     * @param _laterContext The batch context whose time values should come AFTER all of _earlierContext's subsequent queue transactions
+     * @param _firstfirstSubsequentQueueIndex Index of the queue element we are validating came later than the _earlierContext
      * @param _queueRef The storage container for the queue.
      */
-    function _validateContextBeforeEnqueue(
-        BatchContext memory _context,
-        uint40 _queueIndex,
+    function _validateEnqueuesBetweenContexts(
+        BatchContext memory _earlierContext,
+        BatchContext memory _laterContext,
+        uint40 _firstfirstSubsequentQueueIndex,
         iOVM_ChainStorageContainer _queueRef
     )
         internal
         view
     {
-            Lib_OVMCodec.QueueElement memory nextQueueElement = _getQueueElement(
-                _queueIndex,
+            Lib_OVMCodec.QueueElement memory earliestQueueElementBetween = _getQueueElement(
+                _firstfirstSubsequentQueueIndex,
                 _queueRef
             );
 
+            // Assume the earliest and latest are the same, update the latest if not
+            Lib_OVMCodec.QueueElement memory latestQueueElementBetween = earliestQueueElementBetween;
```

Comment:
nit: if this were just a declaration of `Lib_OVMCodec.QueueElement memory latestQueueElementBetween` and we then set `latestQueueElementBetween = earliestQueueElementBetween` in an `else` below it would save a bit of gas.

Reply:
Very nice optimization!